Scripts/CopyClassicVHDFromVMToLab.ps1

```
@@ -0,0 +1,253 @@
+    Write-Host $Done
+
+    $destContainerName = 'uploads'
+    Write-Host "  ensuring destination container '$destContainerName' exists ... " -NoNewline
+    New-AzureStorageContainer -Name $destContainerName -Context $destContext -Permission Off -ErrorAction SilentlyContinue | Out-Null
+    Write-Host $Done
+
+    if ($CopyContext.IsVMDiskManaged)
+    {
+        Write-Host "  copying managed disk ... " -NoNewline
+        $copyHandle = Start-AzureStorageBlobCopy -AbsoluteUri $CopyContext.VMSourceUri -DestContainer $destContainerName -DestBlob $CopyContext.VHDFileName -DestContext $destContext
+    }
+    else
+    {
+        Write-Host "  preparing the source context ... " -NoNewline
+        $srcContext = New-AzureStorageContext -StorageAccountName $CopyContext.VMStorageAccountName -StorageAccountKey $CopyContext.VMStorageAccountKey
+        Write-Host $Done
+
+        Write-Host "  copying unmanaged disk ... " -NoNewline
+        $copyHandle = Start-AzureStorageBlobCopy -AbsoluteUri $CopyContext.VMSourceUri -Context $srcContext -DestContainer $destContainerName -DestBlob $CopyContext.VHDFileName -DestContext $destContext
+    }
+
+    $copyStatus = $copyHandle | Get-AzureStorageBlobCopyState
+    while ($copyStatus.Status -eq "Pending")
+    {
+        $copyStatus = $copyHandle | Get-AzureStorageBlobCopyState 
+        $perComplete = ($copyStatus.BytesCopied / $copyStatus.TotalBytes) * 100
+        Write-Progress -Activity "Copying blob ... " -Status "Percentage Complete" -PercentComplete "$perComplete"
+        Start-Sleep 10
+    }
+
+    Write-Host $Done
+
+    return $copyStatus
+}
+
+###################################################################################################
+#
+# Main execution block.
+#
+
+try
+{
+    Write-Host "Selecting subscription '$SubscriptionId' ... " -NoNewline
+    Select-AzureSubscription -SubscriptionId $SubscriptionId -ErrorAction SilentlyContinue | Out-Null
```

should we silently continue?

---

We used to need it when we had both scripts in one. Now that they are separate, I can change it. In this case, it makes sense that both are correctly set.